sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/compiler/CxxCompilerSensor.java

```diff
@@ -128,20 +128,18 @@ protected void processReport(final Project project, final SensorContext context,
     try {
       parser.processReport(project, context, report, reportCharset, reportRegEx, warnings);
       for (CompilerParser.Warning w : warnings) {
-        // get filename from file system - e.g. VC writes case insensitive file name to html
-        if (isInputValid(w.filename, w.line, w.id, w.msg)) {
+        if (isInputValid(w)) {
           saveUniqueViolation(project, context, parser.rulesRepositoryKey(), w.filename, w.line, w.id, w.msg);
         } else {
-          CxxUtils.LOG.warn("C-Compiler warning: {}", w.msg);
+          CxxUtils.LOG.warn("C-Compiler warning: '{}''{}'", w.id, w.msg);
         }
       }
     } catch (java.io.FileNotFoundException|java.lang.IllegalArgumentException e) {
       CxxUtils.LOG.error("processReport Exception: {} - not processed '{}'", report, e);
     }
   }
 
-  private boolean isInputValid(String file, String line, String id, String msg) {
-    return !StringUtils.isEmpty(file) && !StringUtils.isEmpty(line)
-      && !StringUtils.isEmpty(id) && !StringUtils.isEmpty(msg);
+  private boolean isInputValid(CompilerParser.Warning warning) {
```

Comment:
```isInputValid()``` shall block ```null``` and empty strings. I recommend to check at least id and msg. This 2 strings are also used in line 134 ```CxxUtils.LOG.warn("C-Compiler warning: '{}''{}'", w.id, w.msg);``` and a NullPointerException() will occur if ```msg==null```